### PR TITLE
chore(suite-native): Add basic components for mob app

### PR DESCRIPTION
**Pull request by Nodonisko**
**What inside:**
1. Text, Box, Button components
2. Added fonts to theme
3. Fixes for theme, snippets and some type fixes

**Why is package called `atoms`?**
1. It is just set of "atomic" components
2. Won't we confusing like current status with components inside suite package and standalone components package
3. It is not conflicting with current `components` package, so in future we won't have any problems when we will start creating new v2 components for web
4. Could be easily renamed later if we will change our minds5. 

Simulator Screen Shot - iPhone 13 - 2022-05-08 at 22 12 28

TBD: tests for components

### CODE REVIEW

File: packages/suite-native/src/screens/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    manifest: {
+        email: 'info@trezor.io',
+        appUrl: '@trezor/suite',
+    },
+};
+
+export const Home = () => {
+    const isDarkMode = useColorScheme() === 'dark';
+    const [connectStatus, setConnectStatus] = useState<any>('');
+    const [deviceState, setDeviceState] = useState<any>('');
```
Comment (juriczech) [inline packages/suite-native/src/screens/Home.tsx:28]: Usage and default value indicate that these state properties should be type `string` instead of `any`. Maybe you forgot to fix these typings?

**Review by juriczech — CHANGES_REQUESTED** [review]

File: packages/suite-native/src/screens/Home.tsx (hunk 1 above)
Comment (Nodonisko) [reply packages/suite-native/src/screens/Home.tsx:28]: Would be better, but this page is just some "dev" showcase and this page will be probably deleted soon

**Review by juriczech — APPROVED** [review]

File: packages/styles/src/processStyles.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { mergeStyleObjects, mergeNativeStyleObjects } from './mergeStyleObjects';
 import { Style, NativeStyle, Styles, NativeStyles } from './types';
 
-type BasicRecord = Record<string, unknown>;
+type BasicRecord = object;
```
Comment (dahaca) [inline packages/styles/src/processStyles.ts]: Why did you remove the `Record<>`?

File: packages/atoms/src/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Text as RNText, TextProps as RNTextProps, TextStyle } from 'react-native';
+import { useNativeStyles, prepareNativeStyle, NativeStyleObject } from '@trezor/styles';
+import { Color, TypographyStyle } from '@trezor/theme';
+
+interface TextProps extends Omit<RNTextProps, 'style'> {
```
Comment (dahaca) [inline packages/atoms/src/Text.tsx:6]: I would always keep component prop types right above components like we do now. And styled above those

File: packages/atoms/src/Box.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    'flex',
+    'flexDirection',
+    'justifyContent',
+    'alignItems',
+    'alignContent',
+    'alignSelf',
+] as const;
+
+const spacingStylePropsKeys = [
+    'marginHorizontal',
```
Comment (dahaca) [inline packages/atoms/src/Box.tsx:17]: Looks to be way to flexible ;D That's exactly what @marekrjpolak was removing from modals

File: packages/atoms/src/Text.tsx (hunk 3 above)
Comment (Nodonisko) [reply packages/atoms/src/Text.tsx:6]: I don't see any benefit of doing that, in my opinion it makes more sense to keep prop types always at top. It makes sense because in most cases when you open file you only want to look how you can use component and prop types are kind of docs for component. If you will put prop types above component it will be always in different place because of styles and and other stuff and you will have to find it.

File: packages/atoms/src/Box.tsx (hunk 4 above)
Comment (Nodonisko) [reply packages/atoms/src/Box.tsx:17]: Yea, I had kind of doubts about this, but in the other hand I want to avoid of creating styled component when you need just one margin. This props are intended to use just one or two of them, in case you will need more you should convert it to classic styles.

I am thinking about removing all `layoutStylesPropsKey` or creating eslint rule that will allow you to use maximum 2 of these props until you will have to convert it to styled component.

File: packages/styles/src/processStyles.ts (hunk 2 above)
Comment (Nodonisko) [reply packages/styles/src/processStyles.ts]: I was testing somenthing because I got strange type error, but I switched it back to Record now it seems to works so Record is back now

File: packages/atoms/src/Box.tsx (hunk 4 above)
Comment (dahaca) [reply packages/atoms/src/Box.tsx:17]: Hmmm. Still sketchy. I think that props should be component-related not styles related, with some exception when some prop is used super often. If they are intended to be used extremely often then I get it. Let's see how it pans out :)

File: packages/atoms/src/Text.tsx (hunk 3 above)
Comment (dahaca) [reply packages/atoms/src/Text.tsx:6]: I enjoy when types are next to the function it's used it so everything is grouped. Yes, you have to scroll a bit to find the props but it also would be easy since it's visible where the component-related code starts :D And more often I find myself needing to edit props from within a component than going inside to read them, since it's often possible to preview them with code autocompletion. But again, can't be forcing this alone.

Comment (matejkriz) [reply packages/atoms/src/Text.tsx:6]: I personally also have a slight preference to have component prop types right above component render method, but it doesn't seem important to me.  If you do, I suggest we vote on it in the team and put it in a future "Code style guide"...
